java/java-guestbook/frontend/Dockerfile

```diff
@@ -0,0 +1,26 @@
+# Use maven to compile the java application.
+FROM docker.io/maven AS build-env
+
+# Set the working directory to /app
+WORKDIR /app
+
+# copy the pom.xml file to download dependencies
+COPY pom.xml ./
+
+# download dependencies as specified in pom.xml
+RUN mvn verify --fail-never
+
+# Copy the rest of the working directory contents into the container
+COPY . ./
+
+# Compile the application.
+RUN mvn -Dmaven.test.skip=true package
```

Comment:
same comment as earlier backend, I'm sure you have a good reason you did this, it just seems a bit more than necessary.

Reply:
(ie your creating more layers than you need to)